framework/src/modules/interoperability/base_interoperability_method.ts

```diff
@@ -12,19 +12,43 @@
  * Removal or modification of this copyright notice is prohibited.
  */
 
+import { codec } from '@liskhq/lisk-codec';
+import { utils } from '@liskhq/lisk-cryptography';
 import { BaseMethod } from '../base_method';
 import { BaseInteroperableMethod } from './base_interoperable_method';
 import { NamedRegistry } from '../named_registry';
-import { ImmutableMethodContext, MethodContext } from '../../state_machine';
-import { ChainAccount } from './stores/chain_account';
+import { ImmutableMethodContext, MethodContext, NotFoundError } from '../../state_machine';
+import { ChainAccount, ChainAccountStore } from './stores/chain_account';
 import { CCMsg } from './types';
 import { StoreGetter, ImmutableStoreGetter } from '../base_store';
 import { BaseInteroperabilityStore } from './base_interoperability_store';
+import {
+	EMPTY_BYTES,
+	CCM_SEND_FAILED_CODE_INVALID_FORMAT,
+	CCM_SEND_FAILED_CODE_CHANNEL_UNAVAILABLE,
+	CHAIN_ID_MAINCHAIN,
+	CHAIN_ACTIVE,
+	CCM_SEND_FAILED_CODE_MESSAGE_FEE_EXCEPTION,
+} from './constants';
+import { CcmSendFailEvent } from './events/ccm_send_fail';
+import { CcmSendSuccessEvent } from './events/ccm_send_success';
+import { ccmSchema } from './schemas';
+import { validateFormat } from './utils';
+import { TokenMethod } from '../token';
+import { OwnChainAccountStore } from './stores/own_chain_account';
 
 export abstract class BaseInteroperabilityMethod<
 	T extends BaseInteroperabilityStore
 > extends BaseMethod {
 	protected readonly interoperableCCMethods = new Map<string, BaseInteroperableMethod>();
+	protected _tokenMethod!: TokenMethod & {
```

Comment:
IMO it will be unnecessary type declaration as it will be removed anyways, so if we optimize the code which needs to be removed will seem counterproductive